root/etc/cont-init.d/01-envfile

```diff
@@ -0,0 +1,16 @@
+#! /bin/bash
+
+if [[ "$(ls /var/run/s6/container_environment/ | xargs)" == *"FILE__"* ]]; then
+  for FILENAME in /var/run/s6/container_environment/*; do
+    if [[ "${FILENAME##*/}" == "FILE__"* ]]; then
+      SECRETFILE=$(cat ${FILENAME})
+      if [[ -f ${SECRETFILE} ]]; then
+        FILESTRIP=${FILENAME//FILE__/}
+        cat ${SECRETFILE} > ${FILESTRIP}
```

Comment:
Maybe my testing is not set up properly, but this doesn't seem to actually set the var. 

Log shows
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing...
[env-init] SOMEVAR set from FILE__SOMEVAR
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing...
usermod: no changes

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
We gratefully accept donations at:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    911
User gid:    911
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
```
And if I `sh` in and look at the file, its there:
```
root@nublab_alpinetest.1:/# cat /var/run/s6/container_environment/SOMEVAR
somevalue
```

But `echo "$SOMEVAR"` is empty.

Reply:
That is not exactly how this works, depending on how you exec in you may need to load bash with this wrapper https://github.com/just-containers/s6-overlay/blob/master/builder/overlay-rootfs/usr/bin/with-contenv using: 
```
with-contenv bash
```

All of our scripts and our runtime for the main program pids load env variables with this wrapper (check headers in logic) . https://github.com/linuxserver/docker-mariadb/blob/master/root/etc/services.d/mariadb/run#L1

Reply:
Ok gotcha. Then it mostly seems to work with secrets, but I'm getting a weird additional newline in the variable:

```
root@nublab_alpinetest.1:/# /usr/bin/with-contenv bash
root@nublab_alpinetest.1:/# echo "$SOMEVAR"
somevalue

root@nublab_alpinetest.1:/# echo -n "$SOMEVAR"
somevalue
root@nublab_alpinetest.1:/#
```

Reply:
Might be some shit with cat, I'll dig into it later today.

Reply:
Nah it was my bad. The secret itself had a newline. Works great!